# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/ViewportUtils.h
# Commit: 7782bd4b305a
# Full Hash: 7782bd4b305a61c5945ef715e482973f64339780
# Author: Botond Ballo <botond@mozilla.com>
# Date: 2020-05-06 09:37:16
# Regressor Bug: 1556556
# File Overlap Count: 1
# Description:
#   Bug 1556556 - Add ViewportUtils::IsZoomedContentRoot(). r=kats,mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68727
# ==============================================================================

diff -r 25d6c0b90ad2 -r 7782bd4b305a layout/base/ViewportUtils.h
--- a/layout/base/ViewportUtils.h	Tue May 05 19:22:42 2020 +0000
+++ b/layout/base/ViewportUtils.h	Tue May 05 19:22:49 2020 +0000
@@ -50,6 +50,17 @@
   static nsPoint VisualToLayout(const nsPoint& aPt, PresShell* aContext);
   static nsRect VisualToLayout(const nsRect& aRect, PresShell* aContext);
   static nsPoint LayoutToVisual(const nsPoint& aPt, PresShell* aContext);
+
+  /**
+   * Returns non-null if |aFrame| is inside the async zoom container but its
+   * parent frame is not, thereby making |aFrame| a root of a subtree of
+   * frames representing content that is zoomed in. The value returned in such
+   * cases is the root scroll frame inside the async zoom container.
+
+   * Callers use this to identify points during frame tree traversal where the
+   * visual-to-layout transform needs to be applied.
+   */
+  static const nsIFrame* IsZoomedContentRoot(const nsIFrame* aFrame);
 };
 
 // Forward declare explicit instantiations of GetVisualToLayoutTransform() for
